### PR TITLE
Improve robustness

### PR DESCRIPTION
**Pull request by FR4NK-W**
Improved robustness of signature verification and prepare for removing hard dependency on openssl

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -64,10 +64,11 @@ type Config struct {
 	DNSSD           hinting.DNSHintGeneratorConf    `toml:"dnssd"`
 	MDNS            hinting.MDNSHintGeneratorConf   `toml:"mdns"`
 	Logging         LogConfig                       `toml:"log,omitempty"`
+	CryptoEngine	string                          `toml:"crypto_engine,omitempty"`
 }
 
-func (c Config) WorkingDir() string {
-	return filepath.Join(c.SciondConfigDir, "bootstrapper")
+func (cfg Config) WorkingDir() string {
```
Comment (marcfrei) [inline config/config.go:70]: I'd probably also change the receiver to a pointer type
```
func (cfg *Config) WorkingDir() string {
```

File: fetcher/scion_cppki_verify.go [hunk 2]
```diff
@@ -92,24 +94,26 @@ func verifyTopologySignature(outputPath, workingDir string) error {
...
-func setupVerifyEnv(workingDir string) (ctx context.Context, cancel context.CancelFunc, verifyPath string, err error) {
+func setupVerifyEnv(cfg *config.Config) (ctx context.Context, cancel context.CancelFunc, verifyPath string, err error) {
 	// Signature verification should complete in a timely manner since it is a local operation
 	ctx, cancel = context.WithTimeout(context.Background(), verifyTimeout)
 
-	for _, tool := range []string{"openssl", "scion-pki"} {
-		if _, err = exec.LookPath(tool); err != nil {
-			return
-		}
+	// check 'scion-pki' tool is on path and executable
```
Comment (marcfrei) [inline fetcher/scion_cppki_verify.go:106]:
```
// Check that
```
or
```
// Make sure that
```

File: fetcher/crypto_cmds.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package fetcher
+
+import (
+	"context"
+	"fmt"
+)
+
+// Verify the signature of signedTopology
+// using the CA bundle rootCertsBundlePath, and outputs the verified payload to verifiedTopology.
```
Comment (marcfrei) [inline fetcher/crypto_cmds.go:9]: `and output`

**Review by marcfrei — APPROVED** [review]

File: res/packaging/debian/etc/scion/bootstrapper.toml [hunk 4]
```diff
@@ -1,16 +1,26 @@
...
+        address = "127.0.0.1:8041"
 [dhcp]
         # Whether to enable DHCP discovery or not (default false)
         enable = true
+[dhcpv6]
+    # Whether to enable DHCPv6 discovery or not (default false)
+    enable = true
+[ipv6]
+    # Whether to enable IPv6 Neighbor Discovery Protocol (NDP)
+    # Router Advertisement DNSSL and RDNS discovery or not (default false)
```
[the remark this answers is not in the document]
Comment (FR4NK-W) [reply res/packaging/debian/etc/scion/bootstrapper.toml]: Done

File: config/config.go [hunk 5]
```diff
@@ -64,10 +64,11 @@ type Config struct {
 	DNSSD           hinting.DNSHintGeneratorConf    `toml:"dnssd"`
 	MDNS            hinting.MDNSHintGeneratorConf   `toml:"mdns"`
 	Logging         LogConfig                       `toml:"log,omitempty"`
+	CryptoEngine	string                          `toml:"crypto_engine,omitempty"`
```
[the remark this answers is not in the document]
Comment (FR4NK-W) [reply config/config.go]: Yes, linted.

File: config/config.go (hunk 1 above)
Comment (FR4NK-W) [reply config/config.go]: Done.

File: fetcher/scion_cppki_verify.go (hunk 2 above)
Comment (FR4NK-W) [reply fetcher/scion_cppki_verify.go]: Done.

File: fetcher/crypto_cmds.go (hunk 3 above)
Comment (FR4NK-W) [reply fetcher/crypto_cmds.go]: Done.

**Review by marcfrei — APPROVED** [review]

**Pull request merged**